tldw_Server_API/app/core/Watchlists/pipeline.py

```
@@ -270,10 +279,65 @@ def _record_scraped(
+                    hist_on_304 = bool(history_cfg.get("on_304", False)) if isinstance(history_cfg, dict) else False
+                    try:
+                        hist_per_page = int(history_cfg.get("per_page_limit")) if isinstance(history_cfg.get("per_page_limit"), int) else None
+                    except Exception:
+                        hist_per_page = None
+                    hist_stop_on_seen = bool(history_cfg.get("stop_on_seen", False)) if isinstance(history_cfg, dict) else False
+                    # Load DB seen keys for boundary stop mode
+                    seen_keys: List[str] = []
+                    if hist_stop_on_seen:
+                        try:
+                            seen_keys = db.list_seen_item_keys(int(src.id))
+                        except Exception:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)